### PR TITLE
updating code for pydantic v2

### PR DESCRIPTION
**Pull request by tayllatheodoro**
I only made small changes, where pydantic was called, I updated to pydantic.v1 and I changed the list of dependencies to work with pydantic v2 and current iohub.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -21,7 +21,7 @@ classifiers = [
 # list package dependencies here
 dependencies = [
   "copylot @ git+https://github.com/czbiohub-sf/coPylot",
-  "iohub==0.1.0",
+  "iohub @ git+https://github.com/czbiohub-sf/iohub.git",
```
Comment (edyoshikun) [inline pyproject.toml:24]: I think Talon is writing something but pin this to `0.2.0a1`

Comment (tayllatheodoro) [reply pyproject.toml]: Done

**Review by edyoshikun — APPROVED** [review]
LGTM! I was able to call the `mantis -h` and the `iohub -h` without problems. Thanks @tayllatheodoro

**Comment by talonchandler** [comment]
Looks good, thanks @tayllatheodoro! Merging. 

FYI @ieivanov @edyoshikun I made a very short attempt to fully migrate to pydantic `v2` like we did for `biahub`, and I found that  most of the acquisition settings tests failed. To err on the conservative side and avoid breaking mantis acquisitions, let's merge this and I'll let you or Ed upgrade and test the acquisitions settings on the scope as needed.

**Pull request merged**